### PR TITLE
Mad function implementation

### LINKED ISSUE #1207: mad function implementation

### LINKED ISSUE #1270: mad function implementation

### PR DESCRIPTION
**Pull request by dchigarev**
## What do these changes do?

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1270, Resolves #1207 
- [ ] tests added and passing

### CODE REVIEW

File: modin/pandas/test/test_series.py [hunk 1]
```diff
@@ -1665,9 +1665,8 @@ def test_lt(data):
 
 @pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
 def test_mad(data):
```
Comment (devin-petersohn) [inline modin/pandas/test/test_series.py:1667]: We should also be testing parameters of the `mad` functionality to ensure correct behavior with those.

Comment (dchigarev) [reply modin/pandas/test/test_series.py]: done

**Comment by gshimansky** [comment]
To run (EXPERIMENTAL) Containerized tests successfully please rebase your PR to current master branch.

File: modin/pandas/test/test_dataframe.py [hunk 2]
```diff
@@ -2383,10 +2383,16 @@ def test_lookup(self):
...
             pd.DataFrame(data).lookup([0, 1], ["col1", "col2"])
 
-    def test_mad(self):
-        data = test_data_values[0]
-        with pytest.warns(UserWarning):
-            pd.DataFrame(data).mad()
+    @pytest.mark.parametrize("data", test_data_values)
+    @pytest.mark.parametrize("axis", [None, 0, 1])
+    @pytest.mark.parametrize("skipna", [None, True, False])
+    @pytest.mark.parametrize("level", [0, -1])
```
Comment (devin-petersohn) [inline modin/pandas/test/test_dataframe.py:2389]:
```suggestion
    @pytest.mark.parametrize("level", [0, -1, None])
```

**Pull request merged**